lib/repl.js

```diff
@@ -86,7 +99,10 @@ function REPLServer(prompt, stream, eval_, useGlobal, ignoreUndefined) {
 
   self.useGlobal = !!useGlobal;
   self.ignoreUndefined = !!ignoreUndefined;
-
+  self.useHistory = !!useHistory;
+  if (self.useHistory) {
+    self.historyPath = process.env['NODE_HISTORY_PATH'] || historyPath;
```

Comment:
The repl readme contains examples of how to hook up a REPL to a `net.Socket` for remote inspection – I'm not sure if anyone does this in practice but I feel like it could certainly run afoul of the setuid problem.

A module that's *only* run when the REPL would be started on the CLI, which only handled things like auto-loading of native modules and persistent history, would sidestep these problems.